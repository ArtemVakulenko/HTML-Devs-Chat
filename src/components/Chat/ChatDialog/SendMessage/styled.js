import styled from 'styled-components';

export const StSendMessageContainer = styled.div`
    height: 10%;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
`;
