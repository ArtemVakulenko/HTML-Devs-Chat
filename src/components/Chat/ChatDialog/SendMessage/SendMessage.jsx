import React from 'react';
import PropTypes from 'prop-types';
import { useTheme } from 'src/components/hooks';
import Button from 'src/components/_common/Button';
import { icons } from 'constants';
import { useTranslation } from 'react-i18next';
import { NotificationManager } from 'react-notifications';
import { StSendMessageContainer } from './styled';
import TextArea from '../TextArea/TextArea';

const SendMessage = ({
  match,
  userName,
  sendMessage,
  messageValue,
  setMessageValue }) => {
  const { t } = useTranslation();

  const handleSendMessage = () => {
    const room = match.params.id;
    const date = new Date().getTime();
    const body = {
      author: userName,
      content: messageValue.trim(),
      date,
      room,
    };
    if (!messageValue || /^ *$/.test(messageValue)) {
      NotificationManager.error(t('messageRequired'), t('Error'), 3000);
    }
    sendMessage(body);
  };
  const handleChange = (value) => {
    setMessageValue(value);
  };
  const { colors, theme } = useTheme();

  return (
    <StSendMessageContainer colors={colors} theme={theme}>
      <TextArea placeholder='enterYourMessage' value={messageValue} onChange={handleChange} onKeyDown={handleSendMessage} />
      <Button icon={icons.sendMessage} onClick={handleSendMessage} />
    </StSendMessageContainer>
  );
};

SendMessage.propTypes = {
  messageValue: PropTypes.string,
  setMessageValue: PropTypes.func,
  sendMessage: PropTypes.func,
  match: PropTypes.object,
  userName: PropTypes.string,
  // userPic: PropTypes.string,
};

export default SendMessage;
