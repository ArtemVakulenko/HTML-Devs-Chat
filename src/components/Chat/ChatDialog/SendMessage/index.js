import { connect } from 'react-redux';
import { messageValue, userName, getUserPic } from 'store/mainPage/selectors';
import { setMessageValue, sendMessage } from 'store/mainPage/actions';
import { withRouter } from 'react-router';
import { compose } from 'redux';
import SendMessage from './SendMessage';

const mapStateToProps = (state) => ({
  messageValue: messageValue(state),
  userName: userName(state),
  userPic: getUserPic(state),
});
const mapDispatchToProps = (dispatch) => ({
  setMessageValue: (payload) => dispatch(setMessageValue(payload)),
  sendMessage: (payload) => dispatch(sendMessage(payload)),
});

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withRouter,
)(SendMessage);
