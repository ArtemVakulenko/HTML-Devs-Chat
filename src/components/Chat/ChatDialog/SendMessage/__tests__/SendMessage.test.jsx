import React from 'react';
import SendMessage from '../SendMessage';

jest.mock('react-i18next', () => ({
    useTranslation: jest.fn().mockImplementation(() => ({
      t: jest.fn(),
    })),
  }));
  jest.mock('../../../../hooks', () => ({
    useTheme: jest.fn().mockImplementation(() => ({
      colors: {
        dark: {
          color: '',
          input: '',
          placeholder: '',
        },
      },
      theme: 'dark',
    })),
  }));

  describe('ChatHeader', () => {
    it('Should match snapshot', () => {
      const component = global.shallow(<SendMessage/>);
      expect(component.html()).toMatchSnapshot();
    });
  });
