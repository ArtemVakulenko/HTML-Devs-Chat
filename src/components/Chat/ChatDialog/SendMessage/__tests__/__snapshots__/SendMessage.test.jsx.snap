// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHeader Should match snapshot 1`] = `"<div class=\\"styled__StSendMessageContainer-lnwVcU gYXjlf\\"><textarea class=\\"styled__StTextArea-cYwHWS hmAZYP\\"></textarea><button class=\\"styled__StIcon-chPeWD dNDmRl\\"></button></div>"`;
